### PR TITLE
DDF-3506 Upgrade RESTITO version

### PR DESCRIPTION
**Pull request by ahoffer**
#### What does this PR do?
Upgrades RESTITO to a version that fixes multiple CVEs.

#### Who is reviewing it? 
@mcalcote @garrettfreibott @rzwiefel @vinamartin 

#### Choose 2 committers to review/merge the PR. 
@clockard
@figliold
@rzwiefel
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)
Successful CI build
#### Any background context you want to provide?
Although RESTITO is a test library, it is the source of several CVEs. Rather than including suppressions for the CVEs I would prefer to upgrade the library and eliminate the known vulnerabilities. 

The new version of RESTITO has a new API. As a part of another PR I rewrote the logic to use the new API, but could not include the work. I include the work as part of this PR.

#### What are the relevant tickets?
DDF-3506

#### Checklist:
- [N/A] Documentation Updated
- [N/A] Update / Add Unit Tests
- [N/A] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ahoffer** [comment]
build now

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java [hunk 1]
```diff
@@ -38,6 +40,9 @@
 public class ChunkedContent {
   protected static final Logger LOGGER = LoggerFactory.getLogger(ChunkedContent.class);
 
+  /** Hide the no-arg constructor */
```
Comment (vinamartin) [inline distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java]: Just curious, what is this for?

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 2]
```diff
@@ -308,7 +307,7 @@ public void setup() throws Exception {
       fail("An unrecoverable error occurred from previous test");
     }
 
-    server = new SecureStubServer(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort())).run();
+    server = new StubServer(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort())).run();
```
Comment (shaundmorris) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: ❓  was SecureStubServer no longer needed because of the upgrade?

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java [hunk 3]
```diff
@@ -196,8 +198,9 @@ private ChunkedContentFunction(
       this.headerCapture = headerCapture;
     }
 
+    @SuppressWarnings({"squid:S2629"})
```
Comment (shaundmorris) [inline distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java]: ❓  why did you need the suppression? I assume it is for one of the logging statements in here?

File: distribution/test/itests/test-itests-ddf/pom.xml [hunk 4]
```diff
@@ -132,11 +132,6 @@
             <artifactId>spatial-csw-common</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
```
Comment (shaundmorris) [inline distribution/test/itests/test-itests-ddf/pom.xml]: Nice, should help clarify where restito is actually used.

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java (hunk 1 above)
Comment (ahoffer) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java]: To hide the default constructor. When I did it, I forgot the correct term was "default". So I wrote no argument.
It prevents developers from creating an incompletely initialized instance of the class.

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java (hunk 2 above)
Comment (ahoffer) [reply distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: It was not. Yeats submitted a PR to the project and it was merged. The version we upgraded to includes Yeats's change.

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java (hunk 3 above)
Comment (ahoffer) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java]: Yes. For logging.

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

2018-03-23T07:55:41,176 | ERROR | -global-threads) | DefaultErrorHandler              | rg.apache.camel.util.CamelLogger  204 | 223 - org.apache.camel.camel-core - 2.19.4 | Failed delivery for (MessageId: ID:9502b480-2e6f-11e8-9230-0242ac110003 on ExchangeId: ID-mesos-jenkins-a31e9c1770e347df9cfe25376f2b1f62-linux-large-42278-1521786917257-8-3). Exhausted after delivery attempt: 1 caught: java.lang.Exception: Boom!!!

**Comment by clockard** [comment]
This change set looks like it has some changes not related to this pr. Specifically owasp.

**Comment by ahoffer** [comment]
@clockard I think I'll hold off on this one until the OWASP plugin upgrade is merged. That will straighten out the changes. IIRC, I had to add a suppression for the new version of restito. Looks like it got mixed up with another PR I have going. I probably pushed to the wrong branch.

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java (hunk 3 above)
Comment (paouelle) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/restito/ChunkedContent.java]: ✏️ Can you add the reason next to the squid id?
